static/app/views/issueDetails/actions/index.spec.tsx

```diff
@@ -168,126 +169,123 @@ describe('GroupActions', function () {
-      );
+    const modal = screen.getByRole('dialog');
+    expect(within(modal).getByText('Share Issue')).toBeInTheDocument();
+    expect(updateMock).toHaveBeenCalled();
+  });
 
-      rerender(
+  it('opens delete confirm modal from more actions dropdown', async () => {
+    const org = {
+      ...organization,
+      access: [...organization.access, 'event:admin'],
+    };
+    MockApiClient.addMockResponse({
+      url: `/projects/${org.slug}/${project.slug}/issues/`,
+      method: 'PUT',
+      body: {},
+    });
+    const deleteMock = MockApiClient.addMockResponse({
+      url: `/projects/${org.slug}/${project.slug}/issues/`,
+      method: 'DELETE',
+      body: {},
+    });
+    render(
+      <Fragment>
+        <GlobalModal />
         <GroupActions
-          group={{...group, status: 'resolved'}}
+          group={group}
           project={project}
           organization={org}
           disabled={false}
         />
-      );
+      </Fragment>,
+      {organization: org}
+    );
 
-      const resolvedButton = screen.getByRole('button', {name: 'Resolved'});
-      expect(resolvedButton).toBeInTheDocument();
-      userEvent.click(resolvedButton);
+    userEvent.click(screen.getByLabelText('More Actions'));
+    userEvent.click(await screen.findByTestId('delete-issue'));
 
-      expect(issuesApi).toHaveBeenCalledWith(
-        `/projects/${org.slug}/project/issues/`,
-        expect.objectContaining({data: {status: 'unresolved', statusDetails: {}}})
-      );
+    const modal = screen.getByRole('dialog');
+    expect(
+      within(modal).getByText(/Deleting this issue is permanent/)
+    ).toBeInTheDocument();
+
+    userEvent.click(within(modal).getByRole('button', {name: 'Delete'}));
+
+    expect(deleteMock).toHaveBeenCalled();
+    expect(browserHistory.push).toHaveBeenCalledWith({
+      pathname: `/organizations/${organization.slug}/issues/`,
+      query: {project: project.id},
+    });
+  });
+
+  it('resolves and unresolves issue', () => {
+    const issuesApi = MockApiClient.addMockResponse({
+      url: `/projects/${organization.slug}/project/issues/`,
+      method: 'PUT',
+      body: {...group, status: 'resolved'},
     });
+
+    const {rerender} = render(
+      <GroupActions
+        group={group}
+        project={project}
+        organization={organization}
+        disabled={false}
+      />,
+      {organization}
+    );
+
+    userEvent.click(screen.getByRole('button', {name: 'Resolve'}));
+
+    expect(issuesApi).toHaveBeenCalledWith(
+      `/projects/${organization.slug}/project/issues/`,
+      expect.objectContaining({data: {status: 'resolved', statusDetails: {}}})
+    );
+
+    rerender(
+      <GroupActions
+        group={{...group, status: 'resolved'}}
+        project={project}
+        organization={organization}
+        disabled={false}
+      />
+    );
+
+    const resolvedButton = screen.getByRole('button', {name: 'Resolved'});
+    expect(resolvedButton).toBeInTheDocument();
```

Comment:
since we clicking on the button, we don't need to check if it is present in the UI